src/steps/active-directory/__tests__/converters.test.ts

```diff
@@ -0,0 +1,248 @@
+      createdOn: undefined,
+      displayName: '',
+      name: '',
+      _rawData: [],
+    });
+  });
+});
+
+describe('createMachineEntity', () => {
+  test('properties transferred', () => {
+    const VmMetadata = {
+      vmId: 'string',
+      cloudProvider: 'string',
+      resourceId: 'string',
+      subscriptionId: 'string',
+    };
+    const rawData: Machine = {
+      id: 'string',
+      computerDnsName: 'string',
+      firstSeen: 'string',
+      lastSeen: 'string',
+      classification: 'string',
+      onboardingStatus: 'string',
+      healthStatus: 'string',
+      deviceValue: 'string',
+      agentVersion: 'string',
+      defenderAvStatus: 'string',
+      riskScore: 'string',
+      osPlatform: 'string',
+      osVersion: 'any',
+      osProcessor: 'string',
+      version: 'string',
+      lastIpAddress: 'string',
+      lastExternalIpAddress: 'string',
+      osBuild: 1234,
+      rbacGroupId: 456,
+      rbacGroupName: 'string',
+      exposureLevel: 'string',
+      isAadJoined: true,
+      aadDeviceId: 'any',
+      machineTags: [],
+      osArchitecture: 'string',
+      managedBy: 'string',
+      managedByStatus: 'string',
+      ipAddresses: [],
+      vmMetadata: VmMetadata,
+    };
+    expect(createMachineEntity(rawData)).toEqual({
+      _class: ['HostAgent'],
+      _key: 'string',
+      _rawData: [],
+      _type: 'microsoft_defender_machine',
+      agentVersion: 'string',
+      computerDnsName: 'string',
+      createdOn: undefined,
+      defenderAvStatus: 'string',
+      displayName: 'string',
+      function: ['endpoint-protection', 'vulnerability-detection'],
+      id: 'string',
+      name: 'string',
+      riskScore: 'string',
+    });
+  });
+});
+
+describe('createMachinesDeviceRelationship', () => {
+  test('properties transferred', () => {
+    expect(createMachinesDeviceRelationship(exampleAccountEntity)).toEqual({
+      _class: 'MANAGES',
+      _key: 'microsoft_defender_account_id|manages|FORWARD:_type=user_endPoint:_class=Device/Host:hostname=undefined',
+      _mapping: {
+        relationshipDirection: 'FORWARD',
+        skipTargetCreation: true,
+        sourceEntityKey: 'microsoft_defender_account_id',
+        targetEntity: {
+          _class: ['Device/Host'],
+          _type: 'user_endPoint',
+          category: 'endpoint',
+          displayName: undefined,
+          fqdn: undefined,
+          hostname: undefined,
+          id: undefined,
+          name: 'name',
+          osName: undefined,
+          osVersion: undefined,
+          platform: undefined,
+        },
+        targetFilterKeys: [['_type', '_class', 'hostname']],
+      },
+      _type: 'microsoft_defender_machine_manages_user_endPoint',
+      displayName: 'MANAGES',
+    });
+  });
+});
+const userMock: User = {};
+describe('generateUserKey', () => {
+  test('properties transferred', () => {
+    expect(generateUserKey(userMock)).toEqual(undefined);
+  });
+});
```

Comment:
Is it likely that we'll get user resources that have no id? What is the purpose of this test?